### ISSUE TITLE
Invalid Atom  Syntax when GlobNode Inside Plus/TimesLink

### ISSUE THREAD

**Issue by Eman22S** [opencog/atomspace#2528]
Given the following Link in the atomspace :-
```
 (PlusLink 
        (Number "1")
        (Number "2")
        (Number "3")
        (Number "4"))
````
if you run the following rule using `cog-execute!`
```
(define glob-for-plus
	(BindLink
		(PlusLink
			(Number "1") (Glob "$star") (Number "4"))
		(PlusLink
			(Number "1") (Glob "$star") (Number "4") (Number "5"))))
```
This errror message is returned
```
("cog-execute!" "Invalid Atom syntax: (PlusLink\n  (NumberNode \"1\") ; [2cf08d6d543d00ff][1]\n  (ListLink\n    (NumberNode \"2\") ; [246c4657667bd0d2][1]\n    (NumberNode \"3\") ; [41ad216531059699][1]\n  ) ; [d210afbe142b212f][-1]\n  (NumberNode \"4\") ; [259a93aa3c716a4][1]\n  (NumberNode \"5\") ; [4d904faf1fa8503d][1]\n) ; [d833f562839bf3f0][-1] (/home/eman/atomspace/opencog/atoms/base/ClassServer.cc:138)\nFunction args:\n((BindLink\n   (PlusLink\n      (NumberNode \"1\")\n      (GlobNode \"$star\")\n      (NumberNode \"4\")\n   )\n   (PlusLink\n      (NumberNode \"1\")\n      (GlobNode \"$star\")\n      (NumberNode \"4\")\n      (NumberNode \"5\")\n   )\n)\n)")'
```
The behavior is the same for  Times and Divide Link. Any idea ?

**Comment by linas** [comment]
It's a bug. The arithmetic links never expected to be used with globs/pattern matching.

**Comment by linas** [comment]
fixed in pull req #2529

**Comment by linas** [comment]
closing, create a new issue if there are continuing problems ...
